odoo_module_migrate/squasher.py

```diff
@@ -0,0 +1,80 @@
+#! /usr/bin/env python
+"""
+python script to automate replacing `pick <<hash>> AUTO_COMMIT`
+with `fixup ...`
+"""
+import sys
+import os
+from .log import logger
+import re
+
+p = re.compile(r"^pick [0-9a-z]* <(.+)> (.+)$")
+
+MATCHER = {
+    "transbot@odoo-community.org": [
+        r"OCA Transbot updated translations from Transifex",
+        r"Update translation files",
+        ],
+    "oca+oca-travis@odoo-community.org": [
+        r"^\[UPD\] Update [a-zA-Z_]*.pot",
+        r"Update translation files",
+        ],
+    "oca-git-bot@odoo-community.org": [
+        r"\[UPD\] README.rst",
+        ]
+    }
+
+
+def autosquash(filepath):
+    lines = []
+    with open(filepath, "r") as infile:
+        for line in infile.readlines():
+            m = p.match(line)
+            if m:
+                email, message = m.groups()
+                if any([
+                        re.match(regex, message)
+                        for regex in MATCHER.get(email, [])
+                        ]):
+                    logger.info("Squash commit %s" % line)
+                    lines.append(line.replace("pick", "squash", 1))
```

Comment:
I need to test it but wouldn't this:
```
[IMP] Something
[UPD] Update module.pot
OCA Transbot updated translations from Transifex
OCA Transbot updated translations from Transifex
OCA Transbot updated translations from Transifex
```

result in this?:
```
[IMP] Something
```

What I would expect is this result:
```
[IMP] Something
[UPD] Update module.pot OCA Transbot updated translations from Transifex
```

At least, that's what I normally do when squashing commits.. I group administrative commits together, but I don't mix them with the developer commits

-----

Also I normally reorder lines as well:

Take this:
```
[IMP] Something
[UPD] Update module.pot
[IMP] Added extra feature
OCA Transbot updated translations from Transifex
OCA Transbot updated translations from Transifex
[IMP] Another cool feature
OCA Transbot updated translations from Transifex
```

I try to do this:
```
pick [IMP] Something
pick [IMP] Added extra feature
pick [IMP] Another cool feature
pick [UPD] Update module.pot
squash OCA Transbot updated translations from Transifex
squash OCA Transbot updated translations from Transifex
squash OCA Transbot updated translations from Transifex
```

Reply:
The issue is not in the code but in my interpretation  ! I was squashing administratif commit with the previous one


``` 
Squash administrative commits (if any) with the previous commit for reducing commit noise.
They are named as "[UPD] README.rst", "[UPD] Update $MODULE.pot", "Update translation files"
and similar names, and comes from OCA-git-bot, oca-travis or oca-transbot.
IMPORTANT: Don't squash legit translation commits, authored by their translators,
with the message "Translated using Weblate (...)".
```
I need some help 

![image](https://user-images.githubusercontent.com/1164578/101144069-138c2a00-3618-11eb-94ca-8440d1582ccc.png)

@pedrobaeza what do you think, how we should squash administrative commit ?

Reply:
That's a difficult one, as sometimes the order is not the "real" one depending on the first date stamped on the commits, so I would leave this one not automated and subjected to the contributor criteria. There are already contributors that don't attend this when migrating modules. The idea behind is to have a cleaner commit history, but it's not mandatory.

Reply:
Ok, regarding the best practice.
If I have a translation commit I should try to merge them with other translation commit ? or I should merge it into the previous one (that can be a real commit of code)

Reply:
> If I have a translation commit I should try to merge them with other translation commit ? or I should merge it into the previous one (that can be a real commit of code)

I'm not a git expert, but I think that the first option (merge translation commit) makes more sense.

Reply:
Here's the wiki document with examples: https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests